project/MimaExcludes.scala

```diff
@@ -100,6 +100,30 @@ object MimaExcludes {
           "org.apache.spark.sql.SQLContext.setSession"),
         ProblemFilters.exclude[MissingMethodProblem](
           "org.apache.spark.sql.SQLContext.createSession")
+      ) ++ Seq(
+        // SPARK-10708: Consolidate sort shuffle implementations
```

Comment:
I'm fine with excluding all of `unsafe` but am slightly warier about excluding all of `util.collection` since there are one or two developerAPI classes there.